### DISCUSSION TITLE
ConvertFrom-Network : The term 'Get-AzNetworkWatcher' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is  correct and try again.

### DISCUSSION THREAD

**Discussion by themakira** [PrateekKumarSingh/AzViz#56] [Q&A] [answered]
First... Prateek Singh great job! Fantastic indeed !! Congratulations !! Beautiful to see!!

But I have the following event when handling the command:

PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Export-AzViz -ResourceGroup myresourceinAzure -Verbose -OutputFilePath C:\Users\MyuserinWindows\Downloads\Report\rgvm.png -Show  -Theme dark                 

ConvertFrom-Network : The term 'Get-AzNetworkWatcher' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.

[Log]

Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Install-Mo
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Acquiring providers for assembly: C:\Users\MyuserinWindows\Documents\Win
VERBOSE: Using the provider 'PowerShellGet' for searching packages.
VERBOSE: The -Repository parameter was not specified.  PowerShellGet will use all of t
VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
VERBOSE: The specified Location is 'https://www.powershellgallery.com/api/v2' and Pack
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: Total package yield:'1' for the specified package 'AzViz'.
VERBOSE: Performing the operation "Install-Module" on target "Version '1.1.2' of modul
VERBOSE: The installation scope is specified to be 'CurrentUser'.
VERBOSE: The specified module will be installed in 'C:\Users\MyuserinWindows\Docu
VERBOSE: The specified Location is 'NuGet' and PackageManagementProvider is 'NuGet'.
VERBOSE: Downloading module 'AzViz' with version '1.1.2' from the repository 'https://
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesBy
VERBOSE: InstallPackage' - name='PSGraph', version='2.1.38.27',destination='C:\Users\I
VERBOSE: DownloadPackage' - name='PSGraph', version='2.1.38.27',destination='C:\Users\
uri='https://www.powershellgallery.com/api/v2/package/PSGraph/2.1.38.27'
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/PSGraph/2.1.38.
VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/PSGra
VERBOSE: Completed downloading 'PSGraph'.
VERBOSE: Hash for package 'PSGraph' does not match hash provided from the server.
VERBOSE: InstallPackageLocal' - name='PSGraph', version='2.1.38.27',destination='C:\Us
VERBOSE: InstallPackage' - name='Az.Accounts', version='2.4.0',destination='C:\Users\I
VERBOSE: DownloadPackage' - name='Az.Accounts', version='2.4.0',destination='C:\Users\
uri='https://www.powershellgallery.com/api/v2/package/Az.Accounts/2.4.0'
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/Az.Accounts/2.4
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/Az.Accounts/2.4
                 VERBOSE: Completed downloading 'https://www.powershellgallery.com/api
VERBOSE: Completed downloading 'Az.Accounts'.
VERBOSE: Hash for package 'Az.Accounts' does not match hash provided from the server.
VERBOSE: InstallPackageLocal' - name='Az.Accounts', version='2.4.0',destination='C:\Us
VERBOSE: InstallPackage' - name='Az.Resources', version='4.2.0',destination='C:\Users\
VERBOSE: DownloadPackage' - name='Az.Resources', version='4.2.0',destination='C:\Users
uri='https://www.powershellgallery.com/api/v2/package/Az.Resources/4.2.0'
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/Az.Resources/4.
VERBOSE: Catalog file 'PSGraph.cat' is not found in the contents of the module 'PSGrap
VERBOSE: For publisher validation, current module 'PSGraph' with version '2.1.38.27' w
                  VERBOSE: For publisher validation, using the previously-installed mopublisher name ''. Is this module signed by Microsoft: 'False'.
VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/Az.Re
VERBOSE: Completed downloading 'Az.Resources'.
   VERBOSE: Hash for package 'Az.Resources' does not match hash provided from the serv
VERBOSE: InstallPackageLocal' - name='Az.Resources', version='4.2.0',destination='C:\U
VERBOSE: Installing the dependency module 'PSGraph' with version '2.1.38.27' for the m
VERBOSE: Module 'PSGraph' was installed successfully to path 'C:\Users\IgorLeonardodos
VERBOSE: InstallPackage' - name='AzViz', version='1.1.2',destination='C:\Users\IgorLeo
VERBOSE: DownloadPackage' - name='AzViz', version='1.1.2',destination='C:\Users\IgorLe
                     uri='https://www.powershellgallery.com/api/v2/package/AzViz/1.1.2
VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/AzViz/1.1.2'.
VERBOSE: Catalog file 'Az.Accounts.cat' is not found in the contents of the module 'Az
VERBOSE: Valid authenticode signature found in the file 'Az.Accounts.psd1' for the mod
VERBOSE: For publisher validation, current module 'Az.Accounts' with version '2.4.0' w
  L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: For publisher validation, using the previously-installed module 'Az.Accounts'
'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\Az.Accounts\2.4.0' 
L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: Authenticode issuer 'CN=Microsoft Root Certificate Authority 2011, O=Microsof
matches with the authenticode issuer 'CN=Microsoft Root Certificate Authority 2011, O=
              'Az.Accounts' with version '2.4.0'.
VERBOSE: Installing the dependency module 'Az.Accounts' with version '2.4.0' for the m
VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/AzViz
VERBOSE: Completed downloading 'AzViz'.
VERBOSE: Hash for package 'AzViz' does not match hash provided from the server.
VERBOSE: InstallPackageLocal' - name='AzViz', version='1.1.2',destination='C:\Users\Ig
VERBOSE: Module 'Az.Accounts' was installed successfully to path 'C:\Users\IgorLeonard
VERBOSE: Catalog file 'Az.Resources.cat' is not found in the contents of the module 'A
VERBOSE: Valid authenticode signature found in the file 'Az.Resources.psd1' for the mo
VERBOSE: For publisher validation, current module 'Az.Resources' with version '4.2.0' 
L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: For publisher validation, using the previously-installed module 'Az.Resources
'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\Az.Resources\4.2.0'
L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
VERBOSE: Authenticode issuer 'CN=Microsoft Root Certificate Authority 2011, O=Microsof
'4.2.0' matches with the authenticode issuer 'CN=Microsoft Root Certificate Authority 
'Az.Resources' with version '4.2.0'.
VERBOSE: Installing the dependency module 'Az.Resources' with version '4.2.0' for the 
VERBOSE: Module 'Az.Resources' was installed successfully to path 'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\Az.Resources\4.2.0'.
VERBOSE: Catalog file 'AzViz.cat' is not found in the contents of the module 'AzViz' being installed.
VERBOSE: For publisher validation, current module 'AzViz' with version '1.1.2' with publisher name ''. Is this module signed by Microsoft: 'False'.
VERBOSE: For publisher validation, using the previously-installed module 'AzViz' with 
version '1.1.2' under 'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2' with publisher name ''. Is this
module signed by Microsoft: 'False'.
VERBOSE: Module 'AzViz' was installed successfully to path 'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2'.
PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Import-Module AzViz -Verbose
VERBOSE: Loading module from path 'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2\AzViz.psd1'.
VERBOSE: Populating RepositorySourceLocation property for module AzViz.
VERBOSE: Loading module from path 'C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2\AzViz.psm1'.
VERBOSE: Importing function 'Export-AzViz'.
PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Connect-AzAccount

Account                      SubscriptionName TenantId                             En
                                                                                   vi
                                                                                   ro
                                                                                   nm
                                                                                   en
                                                                                   t 
-------                      ---------------- --------                             --
my@mycompany.com Paid for Use  3ef327d9-65e8-491d-b8d3-4aad2fa4c775 Az

PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Export-AzViz -ResourceGroup myresourceinAzure -Verbose -OutputFilePath C:\Users\MyuserinWindows\Downloads\Report\rgvm.png -Show -LabelVerbosity 2 -Theme dark

    █████╗ ███████╗██╗   ██╗██╗███████╗   
   ██╔══██╗╚══███╔╝██║   ██║██║╚══███╔╝   Author    : Prateek Singh (Twitter @singhprateik)
   ███████║  ███╔╝ ██║   ██║██║  ███╔╝    Module    : Azure Visualizer v1.1.2
   ██╔══██║ ███╔╝  ╚██╗ ██╔╝██║ ███╔╝     Github    : https://github.com/PrateekKumarSingh/AzViz
   ██║  ██║███████╗ ╚████╔╝ ██║███████╗   Document  : https://azviz.readthedocs.io
   ╚═╝  ╚═╝╚══════╝  ╚═══╝  ╚═╝╚══════╝  
 ▶ Testing Graphviz installation... 515ms
    ▶ GraphViz installation path : C:\Program Files\Graphviz\bin\dot.exe
 ▶ Configuring Defaults... 584ms
    ▶  Target Type            : Azure Resource Group
    ▶  Output Format          : png
    ▶  Exluded Resource Types : 
    ▶  Output File Path       : C:\Users\MyuserinWindows\Downloads\Report\rgvm.png
    ▶  Label Verbosity        : 2
    ▶  Category Depth         : 1
    ▶  Sub-graph Direction    : top-to-bottom
    ▶  Theme                  : dark
    ▶  Launch Visualization   : True
 ▶ Target Azure Resource Groups...  609ms
    ▶ myresourceinAzure
 ▶ Starting to generate Azure visualization... 625ms
VERBOSE: Testing Azure login
    ▶ Exporting network associations for resource group: 'myresourceinAzure'
ConvertFrom-Network : The term 'Get-AzNetworkWatcher' is not recognized as the name
of a cmdlet, function, script file, or operable program. Check the spelling of the
name, or if a path was included, verify that the path is correct and try again.
At C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2\src\
private\ConvertTo-DOTLangauge.ps1:28 char:27
+ ... rkObjects = ConvertFrom-Network -TargetType $TargetType -Targets $Tar ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-AzNetworkWatcher:String) [Convert
   From-Network], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,ConvertFrom-Network

PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> The term 'Get-AzNetworkWatcher' is not recognized^C
PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Get-AzNetworkWatcherTopology
Get-AzNetworkWatcherTopology : The term 'Get-AzNetworkWatcherTopology' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that 
the path is correct and try again.
At line:1 char:1
+ Get-AzNetworkWatcherTopology
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-AzNetworkWatcherTopology:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz> Export-AzViz -ResourceGroup myresourceinAzure -Verbose -OutputFilePath C:\Users\MyuserinWindows\Downloads\Report\rgvm.png -Show  -Theme dark                 

    █████╗ ███████╗██╗   ██╗██╗███████╗   
   ██╔══██╗╚══███╔╝██║   ██║██║╚══███╔╝   Author    : Prateek Singh (Twitter @singhprateik)
   ███████║  ███╔╝ ██║   ██║██║  ███╔╝    Module    : Azure Visualizer v1.1.2
   ██╔══██║ ███╔╝  ╚██╗ ██╔╝██║ ███╔╝     Github    : https://github.com/PrateekKumarSingh/AzViz
   ██║  ██║███████╗ ╚████╔╝ ██║███████╗   Document  : https://azviz.readthedocs.io
   ╚═╝  ╚═╝╚══════╝  ╚═══╝  ╚═╝╚══════╝  
 ▶ Testing Graphviz installation... 452ms
    ▶ GraphViz installation path : C:\Program Files\Graphviz\bin\dot.exe
 ▶ Configuring Defaults... 463ms
    ▶  Target Type            : Azure Resource Group
    ▶  Output Format          : png
    ▶  Exluded Resource Types : 
    ▶  Output File Path       : C:\Users\MyuserinWindows\Downloads\Report\rgvm.png
    ▶  Label Verbosity        : 1
    ▶  Category Depth         : 1
    ▶  Sub-graph Direction    : top-to-bottom
    ▶  Theme                  : dark
    ▶  Launch Visualization   : True
 ▶ Target Azure Resource Groups...  488ms
    ▶ myresourceinAzure
 ▶ Starting to generate Azure visualization... 496ms
VERBOSE: Testing Azure login
    ▶ Exporting network associations for resource group: 'myresourceinAzure'
ConvertFrom-Network : The term 'Get-AzNetworkWatcher' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is 
correct and try again.
At C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz\1.1.2\src\private\ConvertTo-DOTLangauge.ps1:28 char:27
+ ... rkObjects = ConvertFrom-Network -TargetType $TargetType -Targets $Tar ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-AzNetworkWatcher:String) [ConvertFrom-Network], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,ConvertFrom-Network

PS C:\Users\MyuserinWindows\Documents\WindowsPowerShell\Modules\AzViz>

[\log]

Would you help me ?

**Comment by themakira** [comment, marked as answer]
There was a solution. I installed Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force and the command was successful.

**Reply by PrateekKumarSingh** [reply]
Thanks for starting this thread @themakira  and providing the answer, I'm adding `Az.Network` module in the required modules in the next release, which will fix this issue without any additional step.

https://github.com/PrateekKumarSingh/AzViz/blob/ca8dba2eedeec7e649dbe812d346c010549d69b3/AzViz/AzViz.psd1#L56

**Reply by slifer225** [reply]
> Thanks for starting this thread @themakira and providing the answer, I'm adding `Az.Network` module in the required modules in the next release, which will fix this issue without any additional step.
> 
> https://github.com/PrateekKumarSingh/AzViz/blob/ca8dba2eedeec7e649dbe812d346c010549d69b3/AzViz/AzViz.psd1#L56

I am using powershell from mac. I encounter the same error complaining on  'Get-AzNetworkWatcher' . I can confirm the repo cloned contains the AZ.Network within the RequiredModeules. Anyway, i able to solve the issue with the solution mentioned by installing the module.
